beacon-chain/sync/service.go

```diff
@@ -66,13 +90,27 @@ func NewSyncService(ctx context.Context, cfg Config, beaconp2p types.P2P, cs typ
 		crystallizedStateBuf:        make(chan p2p.Message, cfg.ActiveStateBufferSize),
 		announceActiveHashBuf:       make(chan p2p.Message, cfg.CrystallizedStateHashBufferSize),
 		activeStateBuf:              make(chan p2p.Message, cfg.CrystallizedStateBufferSize),
+		stateMapping:                make(map[[32]byte]FinalizedBlock),
+		syncMode:                    mode,
+		currentSlotNumber:           0,
 	}
 }
 
 // Start begins the block processing goroutine.
 func (ss *Service) Start() {
 	log.Info("Starting service")
-	go ss.run(ss.ctx.Done())
+
+	switch ss.syncMode {
+	case 0:
+		log.Info("Starting initial sync")
+		go ss.initialSync(ss.ctx.Done())
```

Comment:
do you disagree with this?

Reply:
Initially I thought it might be messy separating them out since the two syncs share a lot of the same fields, but thinking about this again , I can and should separate them. Will make the changes. But checking for if initial sync is required will be moved to the node package instead of here